### DISCUSSION TITLE
What's the purpose of specifically coloring `%s*([_%w]+:)` in Python?

### DISCUSSION THREAD

**Discussion by Jint-lzxy** [ayamir/nvimdots#1290] [Q&A] [answered]
Forgive me for this silly question, but I've just started learning a bit of Python for configuring Kitty and have almost zero prior knowledge of this language 😢 I noticed that in our config, we have [this pattern](https://github.com/ayamir/nvimdots/blob/239df8224fc0f4f2421daa437bb6f87477bd12e3/lua/modules/configs/ui/paint.lua#L13-L17) `%s*([_%w]+:)`, which expands to:
```console
<1> space characters (0 or more repetitions)
<2> alphanumeric characters or underscores (1 or more repetitions)
<3> literal `:`
```
which, in my file, highlights statements such as `else:`. My question is, what's the purpose of specifically highlighting those statements? I've checked #514 and googled it, but can't seem to find a good explanation for this (or maybe I just didn't understand enough and missed something lol)
***
@CharlesChiuGit I saw you attributed those changes. Could you please kindly give me some hints about what this pattern is useful for? Thanks a ton lmao

**Comment by charliie-dev** [comment, marked as answer]
Sorry for the late reply! kinda busy on the weekend.

it's indeed a side effect of setting `%s*([_%w]+:)` for python in `paint.nvim`.
the reason of adding `paint.nvim`: 
> The reason I implemented this is because of the slow performance of [tree-sitter-comment](https://github.com/stsewd/tree-sitter-comment) in large files. Treesitter will inject the comment language for every line comment, which is far from ideal. I've disabled the comment parser, but still wanted to see @something highlighted in Lua comments.[^1]

and `paint.nvim` just grep text pattern instead of using `ts-comment` to detect comments.

In python, we sometime wrote docstrings like so:

```py
def SetDeviceIndex(self, index: int) -> None:
    """
    Set index for device
    
    Parameters:                                             <- notice this line
        index (int): Index of device.
    """
    foo(index)
```
and a screenshot of it:
![image](https://github.com/ayamir/nvimdots/assets/32497323/8d6a4ef2-eb57-43bf-984d-89c51fe38882)

as u can see `Parameters:` in the docstrings does share the same pattern with `else:`, `index:` and `None:`. (quite some false positives lol) 
but I thought this won't do much harm, so i kept that this way. 

[^1]: https://github.com/folke/paint.nvim#-why

**Reply by Jint-lzxy** [reply]
lol thanks for clarifying that! Looks like there's still a lot of stuff I need to grasp on my Python learning journey 😅

**Reply by charliie-dev** [reply]
Or maybe u can have better ideas to eliminate those false positives!
You always pop up some super great ideas! 🫡

**Reply by Jint-lzxy** [reply]
> Or maybe u can have better ideas to eliminate those false positives!

lmao I might be able to come up with something brilliant once I reach ur level of proficiency :D Can't even compile a list of possible false-positives atm lol
